cache/Cache.js

```diff
@@ -0,0 +1,50 @@
+const EventEmitter = require('events')
+const _createRedisClient = require('./util/createRedisClient')
+const msgpack = require('msgpack-lite')
+
+const DEFAULT_EXPIRATION = 86400
+
+class Cache extends EventEmitter {
+  constructor ({ logger, nodes, defaultTtl, redisClientOptions }) {
+    super()
+    this._logger = logger
+    this._defaultTtl = defaultTtl || DEFAULT_EXPIRATION
+
+    const redisClient = this._redisClient =
+      _createRedisClient(nodes, redisClientOptions)
+
+    redisClient.on('ready', () => {
+      this.emit('ready')
+    })
+
+    redisClient.on('error', (err) => {
+      this.emit('error', err)
+    })
```

Comment:
It might be worth listening/logging other events - e.g. 'node error', '+node', '-node', 'reconnecting'.